core/MySigningAtsha204.cpp

```diff
@@ -164,35 +164,35 @@ bool signerAtsha204SignMsg(MyMessage &msg)
 
 	if (DO_WHITELIST(msg.destination)) {
 		// Salt the signature with the senders nodeId and the unique serial of the ATSHA device
-		memcpy(_signing_signing_nonce, &_signing_rx_buffer[SHA204_BUFFER_POS_DATA],
-		       32); // We can reuse the nonce buffer now since it is no longer needed
+		// We can reuse the nonce buffer now since it is no longer needed
+		memcpy(_signing_signing_nonce, _signing_hmac, 32);
 		_signing_signing_nonce[32] = msg.sender;
-		atsha204_getSerialNumber(&_signing_signing_nonce[33]);
+		memcpy(&_signing_signing_nonce[33], _signing_node_serial_info, 9);
 		// We can 'void' sha256 because the hash is already put in the correct place
-		(void)signerSha256(_signing_signing_nonce, 32+1+SHA204_SERIAL_SZ);
-		SIGN_DEBUG(PSTR("Signature salted with serial: %02X%02X%02X%02X%02X%02X%02X%02X%02X\n"),
-		           _signing_signing_nonce[33], _signing_signing_nonce[34], _signing_signing_nonce[35],
-		           _signing_signing_nonce[36], _signing_signing_nonce[37], _signing_signing_nonce[38],
-		           _signing_signing_nonce[39], _signing_signing_nonce[40], _signing_signing_nonce[41]);
+		(void)signerSha256(_signing_signing_nonce, 32+1+9);
+		SIGN_DEBUG(PSTR("SGN:BND:SIG WHI ID=%d\n"), msg.sender);
+		SIGN_DEBUG(PSTR("SGN:BND:SIG WHI SERIAL=%02X%02X%02X%02X%02X%02X%02X%02X%02X\n"),
+		           _signing_node_serial_info[0], _signing_node_serial_info[1], _signing_node_serial_info[2],
+		           _signing_node_serial_info[3], _signing_node_serial_info[4], _signing_node_serial_info[5],
+		           _signing_node_serial_info[6], _signing_node_serial_info[7], _signing_node_serial_info[8]);
 	}
 
 	// Put device back to sleep
 	atsha204_sleep();
 
 	// Overwrite the first byte in the signature with the signing identifier
-	_signing_rx_buffer[SHA204_BUFFER_POS_DATA] = SIGNING_IDENTIFIER;
+	_signing_hmac[0] = SIGNING_IDENTIFIER;
```

Comment:
You are observant :) and yes, it should be ok. The _signing_rx_buffer is still present and used, and _signing_hmac nowdays should just be a pointer into that buffer.